test/reshape_elimination.cpp

```diff
@@ -106,3 +110,310 @@ TEST(reshape_elimination, dot_transpose_to_dot_w_transpose_args)
     ASSERT_EQ(gdot->get_argument(1)->get_argument(0), W);
     ASSERT_EQ(gdot->get_shape(), (Shape{1, 2}));
 }
+
+TEST(reshape_elimination, recurrent_reshapes)
+{
+    Shape shape_a{2, 2, 3, 3, 2, 4};
+    auto generate_func = [shape_a]() {
+        auto A = make_shared<op::Parameter>(element::f32, shape_a);
+        Shape shape_r_1{3, 2, 2, 4, 6};
+        Shape shape_r_2{6, 8, 3, 2};
+        Shape shape_r_3{6, 8, 6};
+        Shape shape_r_4{6, 2, 2, 2, 6};
+        Shape shape_r_5{2, 3, 2, 2, 2, 3, 2};
+        Shape shape_r_6{48, 6};
+
+        auto r_1 = make_shared<op::Reshape>(A, AxisVector{2, 4, 0, 5, 3, 1}, shape_r_1);
+        auto r_2 = make_shared<op::Reshape>(r_1, AxisVector{0, 1, 2, 3, 4}, shape_r_2);
+        auto r_3 = make_shared<op::Reshape>(r_2, AxisVector{0, 1, 2, 3}, shape_r_3);
+        auto r_4 = make_shared<op::Reshape>(r_3, AxisVector{0, 1, 2}, shape_r_4);
+        auto r_5 = make_shared<op::Reshape>(r_4, AxisVector{0, 1, 2, 3, 4}, shape_r_5);
+        auto r_6 = make_shared<op::Reshape>(r_5, AxisVector{0, 1, 2, 3, 4, 5, 6}, shape_r_6);
+
+        auto f = make_shared<Function>(r_6, ParameterVector{A});
+        return f;
+    };
+
+    auto baseline_f = generate_func();
+    auto optimized_f = generate_func();
+    auto baseline_input_shape = baseline_f->get_parameters().at(0)->get_shape();
+
+    pass::Manager pass_manager;
+    pass_manager.register_pass<pass::VisualizeTree>("before_recurrent_reshapes.pdf");
+    pass_manager.register_pass<pass::RecurrentReshapeElimination>();
+    pass_manager.register_pass<pass::ReshapeElimination>();
```

Comment:
Why is `ReshapeElimination` (as opposed to `RecurrentReshapeElimination` being called here? (Same question applies in several other places throughout the tests.)

Reply:
Called `ReshapeElimination` in addition to `RecurrentReshapeElimination` to show how it might work where the fusion creates a `Reshape` that can be eliminated completely by the existing pass. Actually there is just one test which shows that. It is kind of redundant in the other ones. i'll remove them from the other tests

Reply:
Works for me. You might add a comment or something here.